addons/hr_holidays/models/hr_leave.py

```diff
@@ -617,28 +647,10 @@ def create(self, values):
                 holiday_sudo.activity_update()
         return holiday
 
-    def _read(self, fields):
-        if 'name' in fields and 'employee_id' not in fields:
-            fields.add('employee_id')
-        super(HolidaysRequest, self)._read(fields)
-        if 'name' in fields:
-            if self.user_has_groups('hr_holidays.group_hr_holidays_user'):
-                return
-            current_employee = self.env['hr.employee'].sudo().search([('user_id', '=', self.env.uid)], limit=1)
-            for record in self:
-                emp_id = record._cache.get('employee_id') or False
-                if emp_id != current_employee.id:
-                    try:
-                        record._cache['name']
-                        record._cache['name'] = '*****'
-                    except Exception:
-                        # skip SpecialValue (e.g. for missing record or access right)
-                        pass
-
     def write(self, values):
         is_officer = self.env.user.has_group('hr_holidays.group_hr_holidays_user')
 
-        if not is_officer:
+        if not is_officer and not self.env.is_superuser():
```

Comment:
So it won't be possible to change a leave after it began ? Even for HR managers ?

Reply:
This is wrong obviously ;)